redbot/core/commands/converter.py

```diff
@@ -162,140 +185,262 @@ class APIToken(discord.ext.commands.Converter):
+
+        Attributes
+        ----------
+        maximum : Optional[timedelta]
+            If provided, any parsed value higher than this will raise an exception
+        minimum : Optional[timedelta]
+            If provided, any parsed value lower than this will raise an exception
+        allowed_units : Optional[List[str]]
+            If provided, you can constrain a user to expressing the amount of time
+            in specific units. The units you can chose to provide are the same as the
+            parser understands: (``weeks``, ``days``, ``hours``, ``minutes``, ``seconds``)
+        default_unit : Optional[str]
+            If provided, it will additionally try to match integer-only input into
+            a timedelta, using the unit specified. Same units as in ``allowed_units``
+            apply.
+        """
+
+        def __init__(self, *, minimum=None, maximum=None, allowed_units=None, default_unit=None):
+            self.allowed_units = allowed_units
+            self.default_unit = default_unit
+            self.minimum = minimum
+            self.maximum = maximum
+
+        async def convert(self, ctx: "Context", argument: str) -> timedelta:
+            if self.default_unit and argument.isdecimal():
+                delta = timedelta(**{self.default_unit: int(argument)})
+            else:
+                delta = parse_timedelta(
+                    argument,
+                    minimum=self.minimum,
+                    maximum=self.maximum,
+                    allowed_units=self.allowed_units,
+                )
+            if delta is not None:
+                return delta
+            raise BadArgument()  # This allows this to be a required argument.
+
+
+if TYPE_CHECKING:
+
+    def get_timedelta_converter(
+        *,
+        default_unit: Optional[str] = None,
+        maximum: Optional[timedelta] = None,
+        minimum: Optional[timedelta] = None,
+        allowed_units: Optional[List[str]] = None,
+    ) -> Type[timedelta]:
+        ...
+
+
+else:
+
+    def get_timedelta_converter(
+        *,
+        default_unit: Optional[str] = None,
+        maximum: Optional[timedelta] = None,
+        minimum: Optional[timedelta] = None,
+        allowed_units: Optional[List[str]] = None,
+    ) -> Type[timedelta]:
+        """
+        This creates a type suitable for typechecking which works with discord.py's
+        commands.
+        
+        See `parse_timedelta` for more information about how this functions.
+
+        Parameters
+        ----------
+        maximum : Optional[timedelta]
+            If provided, any parsed value higher than this will raise an exception
+        minimum : Optional[timedelta]
+            If provided, any parsed value lower than this will raise an exception
+        allowed_units : Optional[List[str]]
+            If provided, you can constrain a user to expressing the amount of time
+            in specific units. The units you can chose to provide are the same as the
```

Comment:
Not part of PR but since I noticed it, figured I can mention it 🤷‍♂ 
```suggestion
            in specific units. The units you can choose to provide are the same as the
```